### DISCUSSION TITLE
Forbidden landscaping

### DISCUSSION THREAD

**Discussion by telk5093** [OpenTTD/OpenTTD#8888] [Features]
In RollerCoaster Tycoon 2, there is a setting that makes forbidden landscaping (raising, lowering surfaces).

Many OpenTTD users tries to make their rails as straight as possible they can because of the distance & benefits.
In this situation, what if we forbidden landscaping?
We might have to make our rail network more going back around 😂

**Comment by James103** [comment]
A similar effect could be achieved by running OpenTTD with the BaseCosts 5.0 NewGRF active, by setting its "raise/lower land" parameter to the maximum value ("64kx"). This makes even basic landscaping cost over £12 million (in in-game currency), instead of the default £187, forcing players to build their infrastructure to fit the landscape instead of changing the landscape to fit the infrastructure, lest they very quickly run out of money.

**Comment by planetmaker** [comment]
You can already limit landscaping via the config-file-only settings 
terraform_per_64k_frames = 4194304
terraform_frame_burst = 4096
clear_per_64k_frames = 4194304
clear_frame_burst = 4096
tree_per_64k_frames = 4194304
tree_frame_burst = 4096

They limit the long-term permitted terraform/clear/destroy trees and the same in the short-term. I'm not exactly sure anymore how they relate to raising one corner of one tile per tick (or clearing actions / tree destruction). However in greatly reduced settings they become severely restrictive.
You don't want to disallow that completely as even building a bridge head or tunnel or even normal rail on a slope might involve or require one-corner terraforming.

Additionally you can change the cost of landscaping by what James103 wrote.

**Reply by telk5093** [reply]
Your solution, restricting them in config file sounds useful for slowing-down each behaviors, but sounds like not friendly to normal users in my opinion. Isn't it be nice if it is configurable when generating a new map in the Map generation steps, by adding a single flag to savegame format?
Raising costs will be also a solution for it but I think it would be more great to make possible forbid fundamentally, too.
Apart of this, those were useful ideas for me. It gonna be fun if I apply them to a new map someday :)

**Reply by LC-Zorg** [reply]
Limitation of excessive terraforming is used on reddit servers. They're a bit annoying at times, but looking at what some players do or want to do, it's fortunate that these settings exist. :)
On the Map of Poland server, there are no quantitative restrictions on terraformation, but the costs are significantly increased (ca. £200k), thanks to which there is no problem of destroying the map at all. At most, once in a while, someone will spend a billion building a private island offshore. ;)
The complete prevention of changes to the terrain, especially in an online game, would be very limiting and, contrary to appearances, this would not improve the aesthetics of the routes built.

Here is an example from a reddit vanilla server:
![terraforming on reddit server](https://user-images.githubusercontent.com/67662046/112887936-0bc72880-90d4-11eb-9a11-669b394d72fa.png)

From the north you have a winding, natural looking main railway line - to achieve this effect, very large changes in the terrain were needed.
From the south you have a straight route through the peaks of the mountains - here the player has not bothered to change the terrain or adapt to it at all.


> telk5093: Isn't it be nice if it is configurable when generating a new map in the Map generation steps, by adding a single flag to savegame format?

I will definitely agree with that. Maybe not necessarily in the map generation window, but at least in the settings window it would be very useful. But my reason is completely different. :) The point is, there is no easy way to turn off existed restrictions. I suppose it may not be possible to change that at all.

**Reply by planetmaker** [reply]
> Your solution, restricting them in config file sounds useful for slowing-down each behaviors, but sounds like not friendly to normal users in my opinion. Isn't it be nice if it is configurable when generating a new map in the Map generation steps, by adding a single flag to savegame format?

I don't think a single flag cuts it. But it would be an idea to make these 6 settings visible in the settings. It probably belongs to the limitations tab (and explain their actual numbers and behaviour better than their name currently does):

terraform_per_64k_frames = 4194304
terraform_frame_burst = 4096
clear_per_64k_frames = 4194304
clear_frame_burst = 4096
tree_per_64k_frames = 4194304
tree_frame_burst = 4096
